### PR TITLE
Move TaintBasedEvictions feature from alpha to beta (core part).

### PR DESCRIPTION
**Pull request by gyliu513**
**What this PR does / why we need it**:

This is the first patch of https://github.com/kubernetes/features/issues/166 , it is changing name from `alpha` to `beta`

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
Move TaintBasedEvictions feature from alpha to beta.
```

### CODE REVIEW

**Comment by gyliu513** [comment]
Here I have a question for the policy of moving features from alpha to beta: Just rename the annotation from alpha to beta or keep both alpha and beta annotations?

I found an example here by @erictune https://github.com/kubernetes/kubernetes/pull/31026/files , my thinking is that for some internal annotation, we can just rename it; for some annotations used in YAML/JSON template, we may need to copy and keep both alpha and beta, comments?

/cc @gmarek @davidopp

**Comment by jsafrane** [comment]
I know nothing about taints..
/unassing @jsafrane

File: pkg/api/validation/validation_test.go [hunk 1]
```diff
@@ -3642,7 +3642,7 @@ func TestValidatePod(t *testing.T) {
 				Name:      "pod-forgiveness-invalid",
 				Namespace: "ns",
 			},
-			Spec: extendPodSpecwithTolerations(validPodSpec(nil), []api.Toleration{{Key: "node.alpha.kubernetes.io/notReady", Operator: "Exists", Effect: "NoExecute", TolerationSeconds: &[]int64{-2}[0]}}),
+			Spec: extendPodSpecwithTolerations(validPodSpec(nil), []api.Toleration{{Key: "node.beta.kubernetes.io/notReady", Operator: "Exists", Effect: "NoExecute", TolerationSeconds: &[]int64{-2}[0]}}),
```
Comment (gmarek) [inline pkg/api/validation/validation_test.go]: This probably should use `TaintNodeNotReady` constant as a key. Same below.

**Comment by gmarek** [comment]
One nit. Generally looks good.

File: pkg/api/validation/validation_test.go (hunk 1 above)
Comment (gyliu513) [inline pkg/api/validation/validation_test.go]: Done

**Comment by gmarek** [comment]
/lgtm

@davidopp - do you know who approves such changes?

**Comment by davidopp** [comment]
I think we need to wait to merge this. Please see my comments in kubernetes/features#166

**Comment by gmarek** [comment]
@gyliu513 - in the time being I think you can start working on tests. This should actually be the last step of the work.

**Comment by gmarek** [comment]
@gyliu513 - when we can expect PR with changes in tests and documentation?

**Comment by gyliu513** [comment]
@gmarek sorry for late, I saw that this was removed from 1.7 scope, but I will continue work on this to prepare the test and documentation in coming two weeks, hope it is OK.

**Comment by gmarek** [comment]
Sure - I just need to limit number of things that I look at before the freeze. I'm very happy to review changes afterwards.

**Comment by dchen1107** [comment]
Thanks for the pr, and I will review this after 1.7 code freeze since it is out of 1.7 scope

**Comment by gmarek** [comment]
@dchen1107 - I don't think we need to bug you with this one.

**Comment by gmarek** [comment]
@gyliu513 - any progress on tests?

**Comment by gyliu513** [comment]
Synced with @gmarek offline, will work for this in the middle of Aug, sorry for late response.

File: pkg/features/kube_features.go [hunk 2]
```diff
@@ -76,7 +76,7 @@ const (
 	Accelerators utilfeature.Feature = "Accelerators"
 
 	// owner: @gmarek
-	// alpha: v1.6
+	// beta: v1.7
```
Comment (xiangpengzhao) [inline pkg/features/kube_features.go]: It should be 1.8 now :)

**Review by xiangpengzhao — CHANGES_REQUESTED** [review]

**Comment by k82cn** [comment]
We're going to close this one (upgrading to beta) until enough user of this feature.

**Comment by redbaron** [comment]
@k82cn how do you measure usage of this feature? how do you know that enough is enough? Feels like features are promoted very randomly :(

**Pull request closed without merging**